ocean_provider/utils/datatoken.py

```diff
@@ -111,10 +112,31 @@ def verify_order_tx(
             f"Provider was not able to check the signed message in ProviderFees event\n"
         )
 
-    # TO DO - check transfer of providerFeeAmount providerFeeToken to providerFeeAddress
-
+    # check duration
+    if provider_fee_order_log.args.duration > 0:
+        ts = datetime.now().timestamp()
+        if provider_fee_order_log.args.duration < ts:
+            raise AssertionError(
+                f"Validity in transaction exceeds current UTC timestamp"
+            )
     # end check provider fees
 
+    # check if we have an OrderReused event. If so, get orderTxId and switch next checks to use that
+    event_logs = datatoken_contract.events.OrderReused().processReceipt(
+        tx_receipt, errors=DISCARD
+    )
+    order_log = event_logs[0] if event_logs else None
+    if order_log.args.orderTxId:
```

Comment:
I would suggest: if order log and order_log.args.orderTxId, otherwise you will get a "can not get attribute args from None" error